api.go

```diff
@@ -81,6 +81,19 @@ type Client struct {
 
 	// Random seed.
 	random *rand.Rand
+
+	// virtual indicates whether server supports virtual hosted style requests. If not,
+	// default to path style unless it is an Amazon/Google cloud storage endpoint.
+	virtual urlStyle
```

Comment:
The variable name `virtual` is misleading, since it can have all 3 different possible values.:
1. `UknownURLStyle`
2. `PathStyle`
3. `VirtualHostedStylev`

I guess, it is the left-over name when it was a `boolean` in the previous commit. The comment also sounds like the old comment when `virtual` variable was a boolean.

How about `urlStyle` for the variable name?
However, since `urlStyle` is used for the type name, maybe, the type name can be modified as `urlStyleType`... just a suggestion.

As part  of this change, line#274 will also need to be changed, possibly like:
```
func privateNew(endpoint string, creds *credentials.Credentials, secure bool, region string, urlStyle urlStyleType) (*Client, error) {
:
:
	clnt.urlStyle = urlStyle
:
```

Reply:
ok, done.